src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,198 @@
+    });
+
+    addGenomicData: (alterationData: ICBioData[]) => void;
+
+    pathwayHandler: Function;
+
+    constructor(props: IResultsViewPathwayMapperProps){
+        super(props);
+        this.alterationFrequencyData = [];
+        this.isLoading = false;
+        this.activeToasts = [];
+        this.validGenesAccumulator = {};
+    }
+
+    render() {
+
+        // Alteration data of query genes are loaded
+        this.props.store.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+
+            getAlterationData(this.props.store.samples.result,
+                              this.props.store.patients.result,
+                              this.props.store.coverageInformation.result,
+                              this.props.store.sequencedSampleKeysByGene.result!,
+                              this.props.store.sequencedPatientKeysByGene.result!,
+                              this.props.store.selectedMolecularProfiles.result!,
+                              alterationData,
+                              true,
+                              this.alterationFrequencyData,
+                              this.props.store.genes.result!);
+        });
+
+
+        // Alteration data of non-query genes are loaded.
+        const isNewStoreReady = this.storeForAllData &&
+                                this.storeForAllData.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.isComplete &&
+                                this.storeForAllData.samples.isComplete &&
+                                this.storeForAllData.patients.isComplete &&
+                                this.storeForAllData.coverageInformation.isComplete &&
+                                this.storeForAllData.sequencedSampleKeysByGene.isComplete &&
+                                this.storeForAllData.sequencedPatientKeysByGene.isComplete &&
+                                this.storeForAllData.selectedMolecularProfiles.isComplete;
+
+        if(isNewStoreReady){
+
+            this.storeForAllData!.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach( (alterationData, trackIndex) => {
+                getAlterationData(this.storeForAllData!.samples.result,
+                                  this.storeForAllData!.patients.result,
+                                  this.storeForAllData!.coverageInformation.result,
+                                  this.storeForAllData!.sequencedSampleKeysByGene.result!,
+                                  this.storeForAllData!.sequencedPatientKeysByGene.result!,
+                                  this.storeForAllData!.selectedMolecularProfiles.result!,
+                                  alterationData,
+                                  false,
+                                  this.alterationFrequencyData,
+                                  this.props.store.genes.result!);
+                
+            });
+
+            this.addGenomicData(this.alterationFrequencyData);
```

Comment:
presumably this modifies state in some way and should be called in process of rendering.